### PR TITLE
Initial version of zapier integration

### PR DESCRIPTION
**Pull request by tringholm**

### CODE REVIEW

File: zapier/cradlApi.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return makeGetRequest(z, '/models')
+}
+
+async function listTrainings(z, modelId) {
+  return makeGetRequest(z, '/models/' + modelId + '/trainings')
+}
+
+async function listWorkflows(z) {
+    return makeGetRequest(z, '/workflows')
+}
```
Comment (augustak) [inline zapier/cradlApi.js]: Non consistent indentation

File: zapier/creates/executeWorkflow.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    label: 'executeWorkflow',
+    description: 'Post a document to an existing workflow',
+  },
+  operation: {
+    inputFields: [
+      { 
+        key: 'file', 
+        required: true, 
+        type: 'file', 
+        label: 'File'
```
Comment (augustak) [inline zapier/creates/executeWorkflow.js]:
```suggestion
        label: 'File',
```

File: zapier/creates/executeWorkflow.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        key: 'file', 
+        required: true, 
+        type: 'file', 
+        label: 'File' 
+      },
+      { 
+        key: 'workflowId', 
+        required: true, 
+        label: 'Workflow ID',
+        dynamic: 'listWorkflows.id.name'
```
Comment (augustak) [inline zapier/creates/executeWorkflow.js]:
```suggestion
        dynamic: 'listWorkflows.id.name',
```

File: zapier/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+const hydrators = require('./hydrators');
+const listModels = require('./triggers/listModels');
+const listTrainings = require('./triggers/listTrainings');
+const listWorkflows = require('./triggers/listWorkflows');
+const postPrediction = require('./creates/postPrediction');
+const executeWorkflow = require('./creates/executeWorkflow');
```
Comment (augustak) [inline zapier/index.js]: sort

File: zapier/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  ],
+};
+
+const addAuthorization = async (request, z, bundle) => {
+  const basicHash = Buffer.from(
+    `${bundle.authData.app_client_id}:${bundle.authData.app_client_secret}`
+  ).toString(
+    'base64'
+  );
+  const response = await fetch(process.env.API_AUTH_URL, {
```
Comment (augustak) [inline zapier/index.js]: We should consider limiting the scopes fetched in the integration, eg.:

`?scope=api.lucidtech.ai/documents.write,api.lucidtech.ai/predictions.write,api.lucidtech.ai/workflows.executions.write,api.lucidtech.ai/workflows.read,api.lucidtech.ai/models.read,api.lucidtech.ai/trainings.read`

^ might not be 100% correct

File: zapier/test/creates.test.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+const zapier = require('zapier-platform-core');
+
+const App = require('../index');
+const appTester = zapier.createAppTester(App);
+zapier.tools.env.inject();
+
+
+const FILE_URL =
+  'https://cdn.zapier.com/storage/files/f6679cf77afeaf6b8426de8d7b9642fc.pdf';
```
Comment (augustak) [inline zapier/test/creates.test.js]: Is this file supposed to be hardcoded into the code?

File: zapier/test/triggers.test.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        app_client_id: process.env.app_client_id,
+        app_client_secret: process.env.app_client_secret,
+      },
+    };
+    const results = await appTester(
+      App.triggers.listTrainings.operation.perform,
+      bundle
+    );
+  });
+
```
Comment (augustak) [inline zapier/test/triggers.test.js:34]:
```suggestion
```

**Review by augustak — CHANGES_REQUESTED** [review]
Can @hexjelly review the JS code?

File: zapier/cradlApi.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+async function createPrediction(z, documentId, modelId, trainingId){
+  return makePostRequest(z, '/predictions', {
+    documentId: documentId,
+    modelId: modelId,
+    trainingId: trainingId,
+  });
+}
+
+async function executeWorkflow(z, documentId, workflowId){
```
Comment (magnurud) [inline zapier/cradlApi.js]: Don't we want to add keyword-arguments here that is passed to the workflow along with the document? Or is that an advanced feature for later?

File: zapier/index.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+const listWorkflows = require('./triggers/listWorkflows');
+const postPrediction = require('./creates/postPrediction');
+const executeWorkflow = require('./creates/executeWorkflow');
+
+const authentication = {
+  type: 'custom',
+  test: {
+    url: process.env.API_BASE_URL + '/organizations/me',
+  },
+  // If you need any fields upfront, put them here
```
Comment (magnurud) [inline zapier/index.js]: Is this commentary still useful?

**Review by magnurud — CHANGES_REQUESTED** [review]

File: zapier/cradlApi.js (hunk 8 above)
Comment (tringholm) [reply zapier/cradlApi.js]: Do you have any examples of kwargs? We could have a curated list of them, but I don't think free-form kwargs will be looked kindly upon in Zapier's app review when publishing.

File: zapier/test/creates.test.js (hunk 6 above)
Comment (tringholm) [reply zapier/test/creates.test.js]: Yes - it's only used for testing

File: zapier/index.js (hunk 5 above)
Comment (tringholm) [reply zapier/index.js]: Added scopes to the authorization request.

**Review by magnurud — APPROVED** [review]

**Review by augustak — APPROVED** [review]

**Pull request merged**